internal/service/encryptionatrest/resource_encryption_at_rest.go

```diff
@@ -376,9 +371,22 @@ func (r *encryptionAtRestRS) Delete(ctx context.Context, req resource.DeleteRequ
 		return
 	}
 
-	conn := r.Client.Atlas
+	enabled := false
+	connV2 := r.Client.AtlasV2
 	projectID := encryptionAtRestState.ProjectID.ValueString()
-	_, err := conn.EncryptionsAtRest.Delete(ctx, projectID)
+
+	_, _, err := connV2.EncryptionAtRestUsingCustomerKeyManagementApi.GetEncryptionAtRest(context.Background(), projectID).Execute()
```

Comment:
is there a reason we need to do this get request? comparing to previous logic it would seem it does the update directly